WalletWasabi/WabiSabi/Client/CoinJoinManager.cs

```diff
@@ -582,9 +585,32 @@ private static async Task WaitAndHandleResultOfTasksAsync(string logPrefix, para
 		}
 	}
 
-	public void WalletEnteredSendWorkflow(string walletName) => WalletsInSendWorkflow.TryAdd(walletName, 0);
+	public void WalletEnteredSendWorkflow(string walletName) => WalletsInSendWorkflow.TryAdd(walletName, false);
 
-	public void WalletLeftSendWorkflow(string walletName) => WalletsInSendWorkflow.Remove(walletName, out _);
+	public void WalletLeftSendWorkflow(Wallet wallet)
+	{
+		if (CoinJoinClientStates.TryGetValue(wallet.WalletName, out var stateHolder) && WalletsInSendWorkflow.TryRemove(wallet.WalletName, out bool needRestart))
+		{
+			if (needRestart)
+			{
+				Task.Run(async () => await StartAsync(wallet, stateHolder.StopWhenAllMixed, stateHolder.OverridePlebStop, CancellationToken.None).ConfigureAwait(false));
```

Comment:
You can await for it. It is just adding one element to the channel.

Reply:
This method started as async, but got changed because we don't have `async OnNavigatedTo`.

See conversation about it [here](https://github.com/zkSNACKs/WalletWasabi/pull/10500#discussion_r1178792971).

I can revert the change and use Roland's suggestion

```cs
RxApp.MainThreadScheduler.Schedule(async () => await YourMethod());
```

Reply:
https://github.com/zkSNACKs/WalletWasabi/pull/10500#discussion_r1180464727